lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -307,10 +307,23 @@ defmodule Archethic.Mining.PendingTransactionValidation do
          type: :nft,
          data: %TransactionData{content: content}
        }) do
-    if Regex.match?(~r/(?<=initial supply:).*\d/mi, content) do
+    schema =
+      :archethic
+      |> Application.app_dir("priv/json-schemas/token-core.json")
+      |> File.read!()
+      |> Jason.decode!()
+      |> ExJsonSchema.Schema.resolve()
+
+    with {:ok, json_token} <- Jason.decode(content),
+         :ok <- ExJsonSchema.Validator.validate(schema, json_token) do
       :ok
     else
-      {:error, "Invalid NFT content"}
+      :error ->
```

Comment:
To catch `Jason.decode()` error you should use `{:error, %Jason.DecodeError{}}`

Reply:
👍